### PR TITLE
OPSIM-1069: fix m5 diff basis function

### PR DESCRIPTION
**Pull request by yoachim**
Fix bug where M5Diff was using the dark sky brightness rather than the best possible 5-sigma limiting depth. Also update model observatory to create fresh Conditions objects to prevent excessive warnings.

### CODE REVIEW

**Comment by rhiannonlynne** [comment]
Hey, so maybe a stupid question - but why *aren't* we just comparing the current skybrightness maps against the dark skybrightness values? 
This would eliminate the effect of changing seeing resulting in a different overall value for the basis function, and it would still have the desired effect of guiding observations towards the parts of the sky which are darkest and into the filters which would work best at this point in the lunar cycle / position on the sky. 
And would require fewer calculations.

File: rubin_sim/scheduler/model_observatory/model_observatory.py [hunk 1]
```diff
@@ -215,11 +216,13 @@ def __init__(
         sun_moon_info = self.almanac.get_sun_moon_positions(self.mjd)
         season_offset = create_season_offset(self.nside, sun_moon_info["sun_RA"])
         self.sun_ra_start = sun_moon_info["sun_RA"] + 0
+        self.mjd_start = mjd_start
```
Comment (ehneilsen) [inline rubin_sim/scheduler/model_observatory/model_observatory.py:219]: Isn't this already done more reliably in line 121?

**Review by ehneilsen — APPROVED** [review]
Double check line 219 (and/or make sure it behaves properly when mjd_start is None), but otherwise, looks fine.

**Comment by yoachim** [comment]
> Hey, so maybe a stupid question - but why _aren't_ we just comparing the current skybrightness maps against the dark skybrightness values? This would eliminate the effect of changing seeing resulting in a different overall value for the basis function, and it would still have the desired effect of guiding observations towards the parts of the sky which are darkest and into the filters which would work best at this point in the lunar cycle / position on the sky. And would require fewer calculations.

If we only used the dark sky map we would lose the airmass extinction term. By using the full 5-sigma limiting depth, the scheduler "knows" that sticking close to the meridian is much more important in u than in z. 

I think we want the basis function to change value with different seeing conditions. That's a good thing. It means we have to set a fiducial seeing value, which is kind of an arbitrary magic number. But getting that value wrong is only a few tenths of a mag difference, while airmass and skybrightness terms vary by several mags.

File: rubin_sim/scheduler/model_observatory/model_observatory.py (hunk 1 above)
Comment (yoachim) [reply rubin_sim/scheduler/model_observatory/model_observatory.py]: Indeed!

**Pull request merged**